commit 6f0b0c294b995df6c32edc75f90dfac3ad04114a
Author: mjw <mjw@a5019735-40e9-0310-863c-91ae7b9d1cf9>
Date:   Thu Mar 28 22:52:13 2013 +0000

    memcheck/tests/strchr test should not depend on line numbers.
    
    git-svn-id: svn://svn.valgrind.org/valgrind/trunk@13348 a5019735-40e9-0310-863c-91ae7b9d1cf9

diff --git a/memcheck/tests/strchr.stderr.exp3 b/memcheck/tests/strchr.stderr.exp3
index 9f90e69..e1832fa 100644
--- a/memcheck/tests/strchr.stderr.exp3
+++ b/memcheck/tests/strchr.stderr.exp3
@@ -1,9 +1,9 @@
 Conditional jump or move depends on uninitialised value(s)
-   at 0x........: __GI_strchr (mc_replace_strmem.c:211)
+   at 0x........: __GI_strchr (mc_replace_strmem.c:...)
    by 0x........: main (strchr.c:15)
 
 Conditional jump or move depends on uninitialised value(s)
-   at 0x........: __GI_strchr (mc_replace_strmem.c:211)
+   at 0x........: __GI_strchr (mc_replace_strmem.c:...)
    by 0x........: main (strchr.c:15)
 
 Conditional jump or move depends on uninitialised value(s)
diff --git a/memcheck/tests/strchr.vgtest b/memcheck/tests/strchr.vgtest
index e914726..152c97e 100644
--- a/memcheck/tests/strchr.vgtest
+++ b/memcheck/tests/strchr.vgtest
@@ -1,2 +1,3 @@
 prog: strchr
 vgopts: -q
+stderr_filter_args: strchr.c

commit 48b0ae338abe2dcd726e1733532a971837049ceb
Author: mjw <mjw@a5019735-40e9-0310-863c-91ae7b9d1cf9>
Date:   Thu Mar 28 22:52:14 2013 +0000

    For memcheck overlap filter_memcpy str[n]cpy and __GI_str[n]cpy are equal.
    
    git-svn-id: svn://svn.valgrind.org/valgrind/trunk@13349 a5019735-40e9-0310-863c-91ae7b9d1cf9

diff --git a/memcheck/tests/filter_memcpy b/memcheck/tests/filter_memcpy
index 47741ba..737304c 100755
--- a/memcheck/tests/filter_memcpy
+++ b/memcheck/tests/filter_memcpy
@@ -1,5 +1,8 @@
 #! /bin/sh
 
 # mc_replace_strmem.c intercepts various memcpy glibc versions.
+# mc_replace_strmem.c str[n]cpy and __GI_str[n]cpy are the same.
 ./filter_stderr "$@" |
-perl -p -e "s/: memcpy\@\@?GLIBC_[.1-9]+ \(mc_replace_strmem.c:...\)/: memcpy \(mc_replace_strmem.c:...\)/"
+perl -p -e "s/: memcpy\@\@?GLIBC_[.1-9]+ \(mc_replace_strmem.c:...\)/: memcpy \(mc_replace_strmem.c:...\)/" |
+sed -e "s/: __GI_strcpy (mc_replace_strmem.c:/: strcpy (mc_replace_strmem.c:/" |
+sed -e "s/: __GI_strncpy (mc_replace_strmem.c:/: strncpy (mc_replace_strmem.c:/"
